### PR TITLE
[AVFoundation] Update bindings for XCode8. Focus on enums and structs.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/AVFoundation/Enums.cs [hunk 1]
```diff
@@ -211,7 +220,7 @@ public enum AVError : nint {
 		IncompatibleAsset = -11848,
 		FailedToLoadMediaData = -11849,
 		ServerIncorrectlyConfigured = -11850,
-		ApplicationIsNotAuthorizedToUseDevice = 11852,
+		ApplicationIsNotAuthorizedToUseDevice = -11852,
```
Comment (mandel-macaque) [inline src/AVFoundation/Enums.cs]: The above to sign fixes already have a PR on master to fix them asap: https://github.com/xamarin/xamarin-macios/pull/577

**Comment by xamarin-release-manager** [comment]
Build failure

File: src/AVFoundation/AVAudioConverterPrimeInfo.cs [hunk 2]
```diff
@@ -34,5 +34,44 @@ namespace XamCore.AVFoundation {
...
+		{
+			return !left.Equals (right);
+		}
+
+		public override bool Equals(object obj)
+		{
+			if (!(obj is AVAudioConverterPrimeInfo))
+				return false;
+
+			return this.Equals((AVAudioConverterPrimeInfo)obj);
```
Comment (spouliot) [inline src/AVFoundation/AVAudioConverterPrimeInfo.cs]: style, missing space after `(`

File: src/AVFoundation/AVAudioConverterPrimeInfo.cs [hunk 3]
```diff
@@ -34,5 +34,44 @@ namespace XamCore.AVFoundation {
...
+			LeadingFrames = leadingFrames;
+			TrailingFrames = trailingFrames;
+		}
+
+		public override string ToString ()
+		{
+			return $"({LeadingFrames}:{TrailingFrames})";
+		}
+
+		public static bool operator ==(AVAudioConverterPrimeInfo left, AVAudioConverterPrimeInfo right)
```
Comment (spouliot) [inline src/AVFoundation/AVAudioConverterPrimeInfo.cs]: style, missing space before `(`

File: src/AVFoundation/AVAudioConverterPrimeInfo.cs [hunk 4]
```diff
@@ -34,5 +34,44 @@ namespace XamCore.AVFoundation {
...
+		{
+			return left.Equals (right);
+		}
+
+		public static bool operator !=(AVAudioConverterPrimeInfo left, AVAudioConverterPrimeInfo right)
```
Comment (spouliot) [inline src/AVFoundation/AVAudioConverterPrimeInfo.cs]: style, missing space before `(`

File: src/AVFoundation/AVAudioConverterPrimeInfo.cs [hunk 5]
```diff
@@ -34,5 +34,44 @@ namespace XamCore.AVFoundation {
...
+		{
+			return !left.Equals (right);
+		}
+
+		public override bool Equals(object obj)
```
Comment (spouliot) [inline src/AVFoundation/AVAudioConverterPrimeInfo.cs]: style, missing space before `(`

File: src/AVFoundation/AVBeatRange.cs [hunk 6]
```diff
@@ -41,5 +41,38 @@ public AVBeatRange (double startBeat, double lengthInBeats)
 			Start = startBeat;
 			Length = lengthInBeats;
 		}
+
+		public override string ToString ()
+		{
+			return $"(Start={Start},Length={Length})";
+		}
+
+		public static bool operator ==(AVBeatRange left, AVBeatRange right)
```
Comment (spouliot) [inline src/AVFoundation/AVBeatRange.cs]: same style issue on the same methods

File: src/AVFoundation/AVTypes.cs [hunk 7]
```diff
@@ -23,6 +23,7 @@
 
 namespace XamCore.AVFoundation {
 
+	[StructLayout (LayoutKind.Sequential)]
```
Comment (spouliot) [inline src/AVFoundation/AVTypes.cs]: note: fine, but it's the default for the c# compiler

File: src/AVFoundation/Enums.cs [hunk 8]
```diff
@@ -157,6 +157,15 @@ public enum AVAudioSessionInterruptionFlags : nuint_compat_int {
 		ShouldResume = 1
 	}
 
+	[Flags]
+	[NoWatch, NoTV]
+	[Availability (Deprecated = Platform.iOS_6_0)]
+	[Native]
+	public enum  AVAudioSessionSetActiveFlags : uint
```
Comment (spouliot) [inline src/AVFoundation/Enums.cs]: is it needed by some API ?
we don't support iOS < 6 (and Apple does not allow this into the App Store anymore) so I don't see the value of binding it today

File: src/AVFoundation/Enums.cs [hunk 9]
```diff
@@ -655,4 +672,28 @@ public enum AVMovieWritingOptions : nuint
 		TruncateDestinationToMovieHeaderOnly = (1 << 0)
 	}
 #endif
+
+	[NoTV, NoWatch, NoMac, iOS (10,0)]
+	[Native]
+	public enum AVCaptureColorSpace : nint
+	{
+		sRGB = 0,
```
Comment (spouliot) [inline src/AVFoundation/Enums.cs]: -> `Srgb` is what's being used elsewhere

File: src/AVFoundation/Enums.cs [hunk 10]
```diff
@@ -655,4 +672,28 @@ public enum AVMovieWritingOptions : nuint
...
+		P3D65 = 1
+	}
+
+	[iOS (8,0)]
+	[Native]
+	public enum AVMusicTrackLoopCount : nint
+	{
+		AVMusicTrackLoopCountForever = -1
```
Comment (spouliot) [inline src/AVFoundation/Enums.cs]: remove `AVMusicTrackLoop` prefix from member

File: tests/monotouch-test/AVFoundation/AVAudioConverterPrimeInfoTest.cs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+	
+		[Test]
+		public void ConstructorTest ()
+		{
+			uint leading = 2;
+			uint trailing = 30;
+			
+			var info = new AVAudioConverterPrimeInfo (leading, trailing);
+			
+			Assert.AreEqual (leading, info.LeadingFrames);
```
Comment (spouliot) [inline tests/monotouch-test/AVFoundation/AVAudioConverterPrimeInfoTest.cs]: if there's more than one `Assert` in the test method **always** include a message, e.g. the property name `LeadingFrames`. That's useful when stack traces are not available or imprecise.

File: src/AVFoundation/Enums.cs (hunk 8 above)
Comment (mandel-macaque) [inline src/AVFoundation/Enums.cs]: Added for completeness, will remove it.

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by spouliot** [comment]
AVFoundation/AVAudioConverterPrimeInfoTest.cs(4,7): error CS0246: The type or namespace name `AVFoundation' could not be found. Are you missing an assembly reference? 
AVFoundation/AVBeatRangeTest.cs(4,7): error CS0246: The type or namespace name `AVFoundation' could not be found. Are you missing an assembly reference? 
AVFoundation/AVCaptureWhiteBalanceGainsTest.cs(4,7): error CS0246: The type or namespace name `AVFoundation' could not be found. Are you missing an assembly reference? 

^ build failures on watchOS when compiling monotouch-test

This framework is not yet enabled (on watchOS, since it would not build without further changes) so you need to add `#if !WATCH ... #endif` on each test file you add

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by spouliot** [comment]
```[FAIL] AVCaptureWhiteBalanceGainsTest.AreEqualFalseTest : gains1 == gains2```
failed on all (but watchOS) platforms

**Comment by xamarin-release-manager** [comment]
Build success

**Pull request merged**